### PR TITLE
Add ability to set MTU for network interface.

### PR DESCRIPTION
**Pull request by amshinde**
Signed-off-by: Archana Shinde <archana.m.shinde@intel.com>

### CODE REVIEW

File: src/net.c [hunk 1]
```diff
@@ -727,6 +727,36 @@ static int hyper_set_interface_name(struct rtnl_handle *rth,
...
+static int hyper_set_interface_mtu(struct rtnl_handle *rth,
+				int ifindex,
+				unsigned int mtu)
+{
+	struct {
+		struct nlmsghdr n;
+		struct ifinfomsg i;
+		char buf[1024];
+	} req;
+
```
Comment (jcvenegas) [inline src/net.c]: Miss check parameters ? if (index < 0) || !rth

File: src/net.c [hunk 2]
```diff
@@ -727,6 +727,36 @@ static int hyper_set_interface_name(struct rtnl_handle *rth,
...
 }
 
+static int hyper_set_interface_mtu(struct rtnl_handle *rth,
+				int ifindex,
+				unsigned int mtu)
+{
+	struct {
+		struct nlmsghdr n;
+		struct ifinfomsg i;
+		char buf[1024];
```
Comment (jcvenegas) [inline src/net.c]: #define NETLINK_BUFSIZE ?

**Comment by amshinde** [comment]
@mcastelino Can you review this PR?

**Review by mcastelino — COMMENTED** [review]
LGTM

**Comment by mcastelino** [comment]
@amshinde we should still setup the MTU even on the tap side. But things should work even without that.

**Comment by amshinde** [comment]
@mcastelino Sent a PR for the tap side: https://github.com/01org/cc-oci-runtime/pull/645

File: src/net.c [hunk 3]
```diff
@@ -793,6 +826,11 @@ static int hyper_setup_interface(struct rtnl_handle *rth,
 		hyper_set_interface_name(rth, ifindex, iface->new_device_name);
 	}
 
+	if (iface->mtu > 0) {
+		fprintf(stdout, "Setting interface MTU to %d\n", iface->mtu);
+		hyper_set_interface_mtu(rth, ifindex, iface->mtu);
```
Comment (jcvenegas) [inline src/net.c]: Missing check error from hyper_set_interface_mtu

Comment (amshinde) [reply src/net.c]: fixed

File: src/net.c (hunk 1 above)
Comment (amshinde) [reply src/net.c]: fixed

**Comment by sameo** [comment]
LGTM

**Pull request merged**